test/unit/mdc-tabs/mdc-tab-bar.test.js

```diff
@@ -0,0 +1,245 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import bel from 'bel';
+import domEvents from 'dom-events';
+import td from 'testdouble';
+import {createMockRaf} from '../helpers/raf';
+import {supportsCssVariables} from '../../../packages/mdc-ripple/util';
+import {MDCTab} from '../../../packages/mdc-tabs/tab';
+import {MDCTabBar} from '../../../packages/mdc-tabs/tab-bar';
+import {MDCTabBarFoundation} from '../../../packages/mdc-tabs/tab-bar';
+import {cssClasses} from '../../../packages/mdc-tabs/tab/constants';
+
+function getFixture() {
+  return bel`
+    <div>
+      <nav id="basic-tabs" class="mdc-tab-bar">
+        <a class="mdc-tab mdc-tab--active" href="#one">Item One</a>
+        <a class="mdc-tab" href="#two">Item Two</a>
+        <a class="mdc-tab" href="#three">Three</a>
+        <span class="mdc-tab-bar__indicator"></span>
+      </nav>
+    </div>`;
+}
+
+function setupTest() {
+  const fixture = getFixture();
+  const root = fixture.querySelector('.mdc-tab-bar');
+  const indicator = fixture.querySelector('.mdc-tab-bar__indicator');
+  const component = new MDCTabBar(root);
+  return {fixture, root, indicator, component};
+}
+
+suite('MDCTabBar');
+
+test('attachTo returns a component instance', () => {
+  assert.isOk(MDCTabBar.attachTo(setupTest().root) instanceof MDCTabBar);
+});
+
+test('#get tabs returns tabs', () => {
+  const {component} = setupTest();
+
+  assert.isArray(component.tabs);
+
+  for (let i = 0; i < component.tabs.length; i++) {
+    assert.instanceOf(component.tabs[i], MDCTab);
+  }
+});
+
+test('adapter#addClass adds a class to the root element', () => {
+  const {root, component} = setupTest();
+
+  component.getDefaultFoundation().adapter_.addClass('foo');
+  assert.isOk(root.classList.contains('foo'));
+});
+
+test('adapter#removeClass removes a class from the root element', () => {
+  const {root, component} = setupTest();
+  root.classList.add('foo');
+
+  component.getDefaultFoundation().adapter_.removeClass('foo');
+  assert.isNotOk(root.classList.contains('foo'));
+});
+
+if (supportsCssVariables(window)) {
+  test('adapter#bindOnMDCTabSelectedEvent adds event listener for MDCTab:selected on ' +
+       'the component', () => {
+    const {component} = setupTest();
+    const raf = createMockRaf();
+
+    component.getDefaultFoundation().adapter_.bindOnMDCTabSelectedEvent();
+    component.tabs[1].getDefaultFoundation().adapter_.notifySelected();
+
+    raf.flush();
+
+    assert.isTrue(component.tabs_[1].root_.classList.contains(cssClasses.ACTIVE));
+    raf.restore();
+  });
+}
+
+test('adapter#unbindOnMDCTabSelectedEvent removes listener from component', () => {
```

Comment:
I think this test would be more robust if it first added a listener via `bind` and then removed it via `unbind`.